src/dialog-utils.js

```diff
@@ -39,10 +43,52 @@ class DialogUtils extends HTMLElement {
         if (this._observer) this._observer.disconnect();
     }
 
+    onShow(e) {
+        if (e.detail.isModal) {
+            this.disablePageScroll();
+        }
+    }
+
     onClose() {
         this.resetIframes();
+        this.enablePageScroll();
+    }
+
+    /**
+     * Patch showModal() to emit a 'show' event, indicate that the dialog is modal
+     */
+    monkeyPatchShowModalMethod() {
+        const origShowModal = this.dialog.showModal.bind(this.dialog);
+        this.dialog.showModal = (...args) => {
+            const result = origShowModal(...args);
+            this.dialog.dispatchEvent(new CustomEvent('show', {
+                detail: {
+                    isModal: true,
```

Comment:
Instead of monkey-patching, `isModal` can also be determined via the CSS pseudo-selector `dialog:modal`.

```
this.dialog.dispatchEvent(new CustomEvent('show', {
    detail: {
        isModal: this.dialog.id && document.querySelector(`#${this.dialog.id}:modal`) !== null,
    },
    bubbles: true,
}
```